srcpkgs/xbps-triggers/files/dkms

```diff
@@ -21,23 +21,32 @@ UPDATE="$5"
 export PATH="usr/bin:usr/sbin:/usr/sbin:/usr/bin:/sbin:/bin"
 
 remove_modules() {
-	local _modver _kver
+	local _modver _kver _line
 
 	# Remove the specified modules from all kernels.
 	set -- ${dkms_modules}
 	while [ $# -gt 0 ]; do
-		$DKMS status -m "$1" | while read line; do
-			if $(echo "$line" | egrep -vq '(added|built|installed)'); then
+		$DKMS status -m "$1" | while IFS=' ' read _modver _kver _line; do
+			if echo "$_modver $_kver $_line" | grep -Evq '(added|built|installed)'; then
 				shift 2; continue
 			fi
-			_modver=$(echo "$line"|sed "s/$1,[[:blank:]]\([^,]*\)[,:].*/\1/;t;d")
-			_kver=$(echo "$line"|awk '{print $3}'|sed "s/\(.*\),$/\1/")
-			echo -n "Removing DKMS module '${1}-${_modver}' for kernel-${_kver}... "
-			$DKMS remove -m "$1" -v "${_modver}" -k "${_kver}" >/dev/null 2>&1
-			if [ $? -eq 0 ]; then
-				echo "done."
+			_modver=${_modver##*/}; _modver=${_modver%?}; _kver=${_kver%,}
+			if [ "$_kver" = added ] && [ -d "/var/lib/dkms/${1}/${_modver}/build" ]; then
+				echo -n "Cleaning up unbuilt DKMS module '${1}-${_modver}'... "
+				rm -rf "/var/lib/dkms/${1}/${_modver}"
+				if [ $? -eq 0 ]; then
+					echo "done."
+				else
+					echo "FAILED!"
+				fi
```

Comment:
What problem does this fix?

`rm -rf "/var/lib/dkms/$VAR"` is pretty dangerous when the vars haven't been carefully sanitized. If there's ever a format change in the DKMS output, this could break horribly.

Reply:
> What problem does this fix?

See this - https://github.com/void-linux/void-packages/pull/37139#discussion_r873132092

> `rm -rf "/var/lib/dkms/$VAR"` is pretty dangerous

I agree.  

Also, after looking into what causes it, this looks like a `dkms` problem. So, I should probably leave it to them to fix this. Not our job.

_Removed._

Reply:
#### UPDATE:
Nope. This was an bug in the dkms trigger. But this bug existed since a long time ago. Not introduced by this PR.

I've fixed it in the latest commit, and also explained the bug in the commit message.